docs/source-pytorch/fabric/api/fabric_methods.rst

```diff
@@ -187,14 +187,51 @@ the data is written to disk.
 
 .. code-block:: python
 
-    # Download data only on one process
     if fabric.global_rank == 0:
-        download_data("http://...")
+        print("Downloading dataset. This can take a while ...")
+        download_dataset("http://...")
 
-    # Wait until all processes meet up here
+    # All other processes wait here until rank 0 is done with downloading:
     fabric.barrier()
 
-    # All processes are allowed to read the data now
+    # After everyone reached the barrier, they can access the downloaded files:
+    load_dataset()
+
+See also: :doc:`../advanced/distributed_communication`
+
+
+all_gather, all_reduce, broadcast
+=================================
```

Comment:
FYI, this document ("Fabric Methods") is meant as a overview of methods, for users who want to get a quick glance of what Fabric offers. I'm grouping these collectives under one section, to avoid duplication of docs, because the full detail usage with examples is explained in the separate page.